### ISSUE TITLE
Suggest link based on note content (not just note title)

### ISSUE THREAD

**Issue by imed-ghomari** [hadynz/obsidian-sidekick#28]
I don't know if it's in the scope of the plugin, but suggesting links based on words that are found in headings and/or paragraphs  would be awesome :
1. suggest links based on heading and convert them to `[[link#heading]]`
2. suggest links based on paragraphs and convert them to `[[link#^paragraph]]`

**Comment by HEmile** [comment]
I like the suggestion for headings. Maybe less so for paragraphs... Filtering when and when not to suggest links based in paragraphs throughout your vault sounds extremely expensive, and getting you many useless false positives.

**Comment by imed-ghomari** [comment]
> Filtering when and when not to suggest links based in paragraphs throughout your vault sounds extremely expensive, and getting you many useless false positives.

You're right, it would. But nonetheless, I think it should be added as an option in the settings.
In the list of suggestions, the 3 types of links could be distinguished with icons (like it's done in [various complements](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin)) and sorted based on importance :
1. Links to note titles
2. Links to headings
3. Links to paragraphs
